### ISSUE TITLE
React-Dialog Accessibility Suggestions

### ISSUE THREAD

**Issue by t-schemmer** [judgemavi/tiny-bits#1]
Here is a list of some accessibility concerns and suggestions:

- Do not explicitly set `aria-label` on components. It fails the [First Rule of Aria](https://www.w3.org/TR/using-aria/#rule1), furthermore, it's an implementation detail that cannot be determined during component development (i.e. It does not matter what content is placed within this Trigger - the label is always `open dialog`.
https://github.com/judgemavi/tiny-bits/blob/224cd90c59890fcabceaf522408b33a7ccc57ac5/packages/react-dialog/src/index.tsx#L150-L155
https://github.com/judgemavi/tiny-bits/blob/224cd90c59890fcabceaf522408b33a7ccc57ac5/packages/react-dialog/src/index.tsx#L164-L169

 - A dialog can be modal or non-modal. The difference between them is contextual, and out of scope of accessibility. However, behaviour _may_ change between the two, which does affect accessibility: focus order.   Often when a dialog is modal, the keyboard focus is trapped or revolves, essentially the next focus from the last interactive element is placed to the first focusable element, and vice versa.  This does not happen by default in the browser, and therefore **must** be coded; but not so for a non-modal dialog, the default focus sequence remains.
https://github.com/judgemavi/tiny-bits/blob/224cd90c59890fcabceaf522408b33a7ccc57ac5/packages/react-dialog/src/index.tsx#L126

- An additional consideration about modal/non-modal  (although it's not accessibility related)  would be for the API to offer a distinction to open in each.  e.g. replace the property `backdrop` with `modal`, which can be utilized to exec the `showModal()` or `show()` methods.
- The native `dialog` element supports the `ESC` key functionality when used as a modal. However, it is not when it's used as a non-modal, which might confuse some users in some cases; consider adding support for those cases where it might be needed or already afforded.

**Comment by judgemavi** [comment]
Addressed in release 1.0.5
